lighthouse-core/lib/page-functions.js

```diff
@@ -447,19 +447,30 @@ function wrapRequestIdleCallback(cpuSlowdownMultiplier) {
   };
 }
 
-const getNodeDetailsString = `function getNodeDetails(elem) {
+/**
+ * @param {HTMLElement} element
+ */
+function getNodeDetails(element) {
+  const details = {
+    devtoolsNodePath: getNodePath(element),
+    selector: getNodeSelector(element),
+    boundingRect: getBoundingClientRect(element),
+    snippet: getOuterHTMLSnippet(element),
+    nodeLabel: getNodeLabel(element),
+  };
+  window.__nodes = window.__nodes || [];
+  window.__nodes.push({key: details.devtoolsNodePath, node: element});
+  return details;
+}
+
+const getNodeDetailsString = `function getNodeDetails(element) {
```

Comment:
Can we just put `getNodeDetails.toString()` in the exports and remove this?

Reply:
yeah i considered that as well, though this fn is a little different from the rest in that it depends on all the others. so we need those stringified functions inlined here within this one.

when i worked on this with @adrianaixba it seems the most practical to just define this fn as a string and skip the conversion in order to make sure all the deps were available. (otherwise i figured whereever we used getNodeDetails in gatherers, we'd also need to inject each of its dependency fns as well.

i still feel pretty decent this was the right compromise for DX, but if there's another good solution we can def explore it.

Reply:
I had to add non trivial code to this function so I split it out, but am keeping the stringified dep version.